streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -273,13 +273,13 @@ public boolean isRunning() {
     private volatile ThreadMetadata threadMetadata;
     private StreamThread.StateListener stateListener;
 
-    private final Admin adminClient;
     private final ChangelogReader changelogReader;
 
     // package-private for testing
     final ConsumerRebalanceListener rebalanceListener;
     final Consumer<byte[], byte[]> mainConsumer;
     final Consumer<byte[], byte[]> restoreConsumer;
+    final Admin adminClient;
```

Comment:
Why is `adminClient` not final any longer? We still pass it into the `StreamThread` constructor.

Reply:
You mean `private`, right? It is not `private` anymore.
Instead of making the `adminClient` field package private for testing, I would either add a setter `setAdmin()` to `MockClientSupplier` or instantiate the admin in a private field of the `MockClientSupplier` and use the existing getter to set up the admin, or instantiate the admin in a public field of the `MockClientSupplier` and accessing it directly to set it up (similarly to the producer and consumer).

Reply:
Yeah I agree, will move all these variables to private and add getters.